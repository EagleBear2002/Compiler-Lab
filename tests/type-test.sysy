int a(){
    return 1;
}

int a(){
    return 0;
}

int main(){
    b = c();
    int test1;
    int test1;
    int array[2] = {0,0};
    test1 = array;
    test1 = a + array;
    test1 = a(1);
    test1 = test1[0];
    test1();
    a = 1;
    return a;
}
