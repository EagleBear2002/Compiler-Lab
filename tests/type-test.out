Error type 4 at Line 5: Redefined function: a.
Error type 1 at Line 10: Undefined variable: b.
Error type 2 at Line 10: Undefined function: c.
Error type 3 at Line 12: Redefined variable: test1.
Error type 5 at Line 14: type.Type.Type mismatched for assignment.
Error type 6 at Line 15: type.Type.Type mismatched for operands.
Error type 8 at Line 16: Function is not applicable for arguments.
Error type 9 at Line 17: Not an array: test1.
Error type 10 at Line 18: Not a function: test1.
Error type 11 at Line 19: The left-hand side of an assignment must be a variable.
Error type 7 at Line 20: type.Type.Type mismatched for return.
